File: browser-extension/icons/generate-icons.js
const fs = require('fs');
const path = require('path');

// Create a simple PNG icon using Canvas (if available) or fallback to a base64 encoded PNG
// This creates a simple colored square as a placeholder

// Simple 1x1 purple pixel PNG (base64)
const purplePixelBase64 = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8DwHwAFBQIAX8jx0gAAAABJRU5ErkJggg==';

function createIcon(size) {
  // For simplicity, we'll create a solid color square
  // In production, you'd want to use a proper canvas library
  
  const canvas = `
<!DOCTYPE html>
<html>
<body>
<canvas id="c" width="${size}" height="${size}"></canvas>
<script>
const c = document.getElementById('c');
const ctx = c.getContext('2d');
const gradient = ctx.createLinearGradient(0, 0, ${size}, ${size});
gradient.addColorStop(0, '#667eea');
gradient.addColorStop(1, '#764ba2');
ctx.fillStyle = gradient;
ctx.fillRect(0, 0, ${size}, ${size});
ctx.fillStyle = 'white';
ctx.font = 'bold ${Math.floor(size/2)}px Arial';
ctx.textAlign = 'center';
ctx.textBaseline = 'middle';
ctx.fillText('B', ${size/2}, ${size/2});
const dataUrl = c.toDataURL();
console.log('DATA:' + dataUrl);
</script>
</body>
</html>`;
  
  return canvas;
}

console.log('Icon generation script');
console.log('=====================');
console.log('');
console.log('To generate icons, please use one of these methods:');
console.log('');
console.log('1. Open icon-generator.html in your browser (EASIEST)');
console.log('2. Use an online tool: https://cloudconvert.com/svg-to-png');
console.log('3. Use image editing software');
console.log('');
console.log('Required files:');
console.log('  - icon16.png (16x16)');
console.log('  - icon48.png (48x48)');
console.log('  - icon128.png (128x128)');

File: browser-extension/icons/generate_icons.py
"""
BookShare Extension Icon Generator
Generates PNG icons from base64 data
"""

import base64
from pathlib import Path

# Simple placeholder icons (solid colors with text)
# These are minimal PNGs that will work as placeholders

def create_placeholder_icon(size, output_path):
    """Create a simple placeholder PNG icon"""
    
    # Create a minimal PNG with ImageMagick command
    import subprocess
    
    try:
        # Try using ImageMagick if available
        cmd = [
            'magick',
            '-size', f'{size}x{size}',
            'xc:#667eea',
            '-font', 'Arial',
            '-pointsize', str(size // 2),
            '-fill', 'white',
            '-gravity', 'center',
            '-annotate', '+0+0', 'B',
            output_path
        ]
        subprocess.run(cmd, check=True)
        print(f"✓ Created {output_path} using ImageMagick")
        return True
    except (subprocess.CalledProcessError, FileNotFoundError):
        print(f"✗ ImageMagick not found. Please use the HTML generator instead.")
        return False

def main():
    print("BookShare Icon Generator")
    print("=" * 50)
    print()
    
    # Icon sizes needed
    sizes = [16, 48, 128]
    icons_dir = Path(__file__).parent
    
    print(f"Icons directory: {icons_dir}")
    print()
    
    success_count = 0
    for size in sizes:
        output_path = icons_dir / f"icon{size}.png"
        if create_placeholder_icon(size, str(output_path)):
            success_count += 1
    
    print()
    if success_count == 0:
        print("⚠️  Automatic generation failed.")
        print()
        print("Please use one of these methods instead:")
        print()
        print("Method 1 (EASIEST): Open icon-generator.html in your browser")
        print(f"  File: {icons_dir / 'icon-generator.html'}")
        print()
        print("Method 2: Use online converter")
        print("  1. Go to: https://cloudconvert.com/svg-to-png")
        print(f"  2. Upload: {icons_dir / 'icon128.svg'}")
        print("  3. Convert to 128x128, 48x48, 16x16")
        print("  4. Save as icon128.png, icon48.png, icon16.png")
        print()
    else:
        print(f"✓ Successfully created {success_count}/3 icons!")

if __name__ == '__main__':
    main()
